ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/OfferingToStateMapper.kt

```diff
@@ -14,19 +14,25 @@ internal fun Offering.toPaywallViewState(
 ): PaywallViewState {
     val paywallData = this.paywall
         ?: return PaywallViewState.Error("No paywall data for offering: $identifier")
-    return try {
-        val templateConfiguration = TemplateConfigurationFactory.create(
-            variableDataProvider = variableDataProvider,
-            mode = mode,
-            paywallData = paywallData,
-            packages = availablePackages,
-            activelySubscribedProductIdentifiers = emptySet(), // TODO-PAYWALLS: Check for active subscriptions
-        )
-        PaywallViewState.Loaded(
-            templateConfiguration = templateConfiguration,
-            selectedPackage = templateConfiguration.packages.default,
-        )
-    } catch (e: Exception) {
-        PaywallViewState.Error("Error creating paywall: ${e.message}")
-    }
+
+    return toPaywallViewState(variableDataProvider, paywallData, mode)
+}
+
+internal fun Offering.toPaywallViewState(
+    variableDataProvider: VariableDataProvider,
+    paywallData: PaywallData,
+    mode: PaywallViewMode,
+): PaywallViewState.Loaded {
```

Comment:
@vegaro this is what I was telling you about :) thanks to your refactor in #1273 (I'll wait for that to be finished before merging this) we can create a `PaywallViewState.Loaded` here since we can guarantee we have everything ready (useful for the default / loading paywalls).